base/cvd/cuttlefish/host/commands/cvd/fetch/fetch_cvd.cc

```diff
@@ -314,7 +316,7 @@ Result<std::vector<std::string>> FetchSystemImgZipImages(
       system_img_zip_name);
 }
 
-Result<BuildApi> GetBuildApi(const BuildApiFlags& flags) {
+Result<CachingBuildApi> GetBuildApi(const BuildApiFlags& flags) {
```

Comment:
nit: It seems you made `CachingBuildApi` inherit from `BuildApi`, so here you could return `std::unique_ptr<BuildApi>` and use the high level interface everywhere else.